docs/src/main/asciidoc/funqy-knative-events.adoc

```diff
@@ -0,0 +1,363 @@
+and `annotated` Cloud Event source.
+
+So, if look at all the functions defined within `SimpleFunctionChain` you'll notice that one function triggers the next.
+The last function that is triggered is `lastChainLink`.
+
+[source, java]
+----
+import io.quarkus.funqy.Context;
+import io.quarkus.funqy.Funq;
+
+public class SimpleFunctionChain {
+    @Funq
+    public void lastChainLink(String input, @Context CloudEvent event) {
+        log.info("*** lastChainLink ***");
+        log.info(input + "::" + "lastChainLink");
+    }
+}
+----
+
+There are two things to notice about this function.  One, it has no output.  Your functions are not
+required to return output.  Second, there is an additional `event` parameter to the function.
+
+If you want to know additional information about the incoming Cloud Event, you can inject the
+`CloudEvent` interface using the Funqy `@Context` annotation.  The `CloudEvent` interface exposes information
+about the triggering event.
+
+[source, java]
+----
+public interface CloudEvent {
+    String id();
+    String specVersion();
+    String source();
+    String subject();
+    OffsetDateTime time();
+}
+----
+
+== Maven
+
+If you look at the {quickstarts-tree-url}/funqy-quickstarts/funqy-knative-events-quickstart/pom.xml[pom],
+you'll see that it is a typical Quarkus pom that pulls in one funqy dependency
+
+[source, xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-funqy-knative-events</artifactId>
+</dependency>
+----
+
+== Dev mode and Testing
+
+Funqy Knative Events support dev mode and unit testing using RestAssured.  You can invoke on Funqy Knative Events functions
+using the same invocation model as
+link:funqy-http[Funqy HTTP] using normal HTTP requests, or Cloud Event Binary mode, or Structured Mode.  All
+invocation modes are supported at the same time.
+
+So, if you open up the unit test code in {quickstarts-tree-url}/funqy-quickstarts/funqy-knative-events-quickstart/src/test/java/org/acme/funqy/FunqyTest.java[FunqyTest.java]
+you'll see that its simply using RestAssured to make HTTP invocations to test the functions.
+
+Funqy also works with Quarkus Dev mode!
+
+== Build the Project
+
+First build the Java artifacts:
+
+[source, shell]
+----
+mvn clean install
+----
+
+Next, a docker image is required by Knative, so you'll need to build that next:
+
+[source, shell]
+----
+docker build -f src/main/docker/Dockerfile.jvm -t yourAccountName/funqy-knative-events-quickstart .
+----
+
+Make sure to replace `yourAccountName` with your docker or quay account name when you run `docker build`.  The
+Dockerfile is a standard Quarkus dockerfile.  No special Knative magic.
+
+Push your image to docker hub or quay
+
+[source, shell]
+----
+docker push yourAccountName/funqy-knative-events-quickstart
+----
+
+Again, make sure to replace `yourAccountName` with your docker or quay account name when you run `docker push`.
+
+== Deploy to Kubernetes/Openshift
```

Comment:
In a subsequent revisions, we should see how we can integrate this with the Quarkus Kubernetes extensions that has some support for KNative